File: leave-management-service/src/main/java/company/Leave_Management_service/Models/LeaveRequest.java
package company.Leave_Management_service.Models;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
public class LeaveRequest {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID id;

    private String employeeId;

    private String username;

    private LocalDate startDate;

    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    private LeaveType type;

    @Enumerated(EnumType.STRING)
    private LeaveStatus status;

    private String reason;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: src/app/Models/LeaveRequest.ts
export interface LeaveRequest {
    id: string;
    employeeId: string;
    username:string;
    startDate: string;
    endDate: string;
    type: LeaveType;
    status: LeaveStatus;
    reason: string;
    createdAt: string;
    updatedAt: string;
  }
  
  export enum LeaveType {
    VACATION = 'VACATION',
    SICK = 'SICK',
    PERSONAL = 'PERSONAL',
    MATERNITY = 'MATERNITY',
    PATERNITY = 'PATERNITY',
    BEREAVEMENT = 'BEREAVEMENT'
  }
  
  export enum LeaveStatus {
    PENDING = 'PENDING',
    APPROVED = 'APPROVED',
    REJECTED = 'REJECTED',
    CANCELLED = 'CANCELLED'
  }
  
  export interface LeaveRequestDto {
    startDate: string;
    endDate: string;
    type: LeaveType;
    reason: string;
  }